arctic/serialization/numpy_records.py

```diff
@@ -91,7 +134,7 @@ class PandasSerializer(object):
     def _index_to_records(self, df):
         metadata = {}
         index = df.index
-        index_tz = None
+        index_tz: Optional[Union[str, List[str]]]
```

Comment:
the variable is not created, should it be created to None?

`index_tz: Optional[Union[str, List[str]]] = None`

Reply:
`index_tz` is guaranteed to be set by the following code. The type hint here just declares the type of the variable.

> ```py
> # You don't need to initialize a variable to annotate it
> a: int  # Ok (no value at runtime until assigned)
> ```
https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html

Reply:
Just spotted it does need a small correction though to:
```py
index_tz: Union[Optional[str], List[Optional[str]]]
```

as if it is a `List`, the entries can be `None` as well.